contracts/VaultFactory.sol

```diff
@@ -29,6 +32,9 @@ contract VaultFactory is ProxyFactory {
       bTokenImplementation = _bTokenImplementation;
       // first project added has id 0
       _nextProjectId = 0;
+      // set up permissions
+      _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
```

Comment:
`_setupRole` initialises the DEFAULT_ADMIN_ROLE as well as any customised roles we make. the difference between `_setupRole` and `grantRole` is that `grantRole` checks first to see if the caller is an admin, which is it's recommended that `_setupRole()` *only* be called in the constructor, otherwise yes it does circumvent the system

`_setupRole` calls the private function `_grantRole` which gives gives the caller that role